### ISSUE TITLE
fails on any valid line, complaining on closing

### ISSUE THREAD

**Issue by Ryder17z** [pascalre/vscode-yaml-sort#6]
this was the only reason why downloaded vscode in the first place...

**Comment by pascalre** [comment]
@Ryder17z can you provide an example document, which causes the plugin to fail?

**Comment by Ryder17z** [comment]
`{
    "Objects": {
        "Object": [
            {
                "Duration": {
                    "_numTurns": "-1"
                },
                "EnemyTag": {
                    "_text": "Enemy Creatures cost 1 less credit to play."
                },
                "GlobalModifiers": {
                    "Self": {
                        "Stat": {
                            "_amount": "-1",
                            "_type": "creatureCostModifier"
                        }
                    }
                },
                "PlayerTag": {
                    "_text": "Your Creatures cost 1 less credit to play."
                },
                "Sounds": {
                    "Play": {
                        "_name": "blip1"
                    }
                },
                "_artist": "Glen Smith",
                "_ObjectType": "Condition",
                "_cost": "0",
                "_subTypes": ""
            },
            {
                "Description": {
                    "_text": "Your Creatures cost 1 less credit to play."
                },
                "Duration": {
                    "_numTurns": "3"
                },
                "GlobalModifiers": {
                    "Self": {
                        "Stat": {
                            "_amount": "-1",
                            "_type": "creatureCostModifier"
                        }
                    }
                },
                "Sounds": {
                    "Play": {
                        "_name": "blip1"
                    }
                },
                "_artist": "Glen Smith",
                "_ObjectType": "Support",
                "_cost": "1",
                "_subTypes": "Deception"
            },
            {
                "Description": {
                    "_text": "Armor 1"
                },
                "Flavour": {
                    "_text": ""
                },
                "Sounds": {
                    "Destroyed": {
                        "_name": "power_down1"
                    },
                },
                "Stats": {
                    "Creature": {
                        "Stat": [
                            {
                                "_amount": "8",
                                "_type": "maxHealth"
                            },
                            {
                                "_amount": "4",
                                "_type": "attack"
                            },
                            {
                                "_amount": "1",
                                "_type": "armor"
                            }
                        ]
                    }
                },
                "_artist": "Glen Smith",
                "_ObjectType": "Agent",
                "_cost": "2",
                "_subTypes": "Mech"
            },
            {
                "Flavour": {
                    "_text": "Sam's got your back."
                },
                "Sounds": {
                    "Ability": {
                        "_name": "pulse1"
                    },
                },
                "Stats": {
                    "Creature": {
                        "Stat": [
                            {
                                "_amount": "11",
                                "_type": "maxHealth"
                            },
                            {
                                "_amount": "5",
                                "_type": "attack"
                            }
                        ]
                    }
                },
                "_artist": "Glen Smith",
                "_ObjectType": "Agent",
                "_cost": "3",
                "_subTypes": "Witch"
            }
        ]
    }
}`

**Comment by Ryder17z** [comment]
```
{
    "Objects": {
        "Object": [
            {
                "Duration": {
                    "_numTurns": "-1"
                },
                "EnemyTag": {
                    "_text": "Enemy Creatures cost 1 less credit to play."
                },
                "GlobalModifiers": {
                    "Self": {
                        "Stat": {
                            "_amount": "-1",
                            "_type": "creatureCostModifier"
                        }
                    }
                },
                "PlayerTag": {
                    "_text": "Your Creatures cost 1 less credit to play."
                },
                "Sounds": {
                    "Play": {
                        "_name": "blip1"
                    }
                },
                "_artist": "Glen Smith",
                "_ObjectType": "Condition",
                "_cost": "0",
                "_subTypes": ""
            },
            {
                "Description": {
                    "_text": "Your Creatures cost 1 less credit to play."
                },
                "Duration": {
                    "_numTurns": "3"
                },
                "GlobalModifiers": {
                    "Self": {
                        "Stat": {
                            "_amount": "-1",
                            "_type": "creatureCostModifier"
                        }
                    }
                },
                "Sounds": {
                    "Play": {
                        "_name": "blip1"
                    }
                },
                "_artist": "Glen Smith",
                "_ObjectType": "Support",
                "_cost": "1",
                "_subTypes": "Deception"
            },
            {
                "Description": {
                    "_text": "Armor 1"
                },
                "Flavour": {
                    "_text": ""
                },
                "Sounds": {
                    "Destroyed": {
                        "_name": "power_down1"
                    },
                },
                "Stats": {
                    "Creature": {
                        "Stat": [
                            {
                                "_amount": "8",
                                "_type": "maxHealth"
                            },
                            {
                                "_amount": "4",
                                "_type": "attack"
                            },
                            {
                                "_amount": "1",
                                "_type": "armor"
                            }
                        ]
                    }
                },
                "_artist": "Glen Smith",
                "_ObjectType": "Agent",
                "_cost": "2",
                "_subTypes": "Mech"
            },
            {
                "Flavour": {
                    "_text": "Sam's got your back."
                },
                "Sounds": {
                    "Ability": {
                        "_name": "pulse1"
                    },
                },
                "Stats": {
                    "Creature": {
                        "Stat": [
                            {
                                "_amount": "11",
                                "_type": "maxHealth"
                            },
                            {
                                "_amount": "5",
                                "_type": "attack"
                            }
                        ]
                    }
                },
                "_artist": "Glen Smith",
                "_ObjectType": "Agent",
                "_cost": "3",
                "_subTypes": "Witch"
            }
        ]
    }
}
```

**Comment by Ryder17z** [comment]
When this file is converted to yaml, your sorting tool does not want to deal with it.

**Comment by pascalre** [comment]
@Ryder17z can you provide the converted yaml?

**Comment by Ryder17z** [comment]
Sorry, I lost it. the other file that also failed was this one:
[sample.yml.txt](https://github.com/pascalre/vscode-yaml-sort/files/3855855/sample.yml.txt)

**Comment by pascalre** [comment]
@Ryder17z the delimiters in the given yaml cause the problem. Currently it is not possible to use delimiters with comments or other text in the same line. I will address this issue in the branch [issue-#6](https://github.com/pascalre/vscode-yaml-sort/tree/issue-%236).

**Comment by pascalre** [comment]
@Ryder17z problem is solved in version 2.3.0
